contracts/HubPool.sol

```diff
@@ -482,9 +484,12 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
         // If they try access more funds that available (i.e l1TokensToReturn > liquidReserves) this will underflow.
         pooledTokens[l1Token].liquidReserves -= l1TokensToReturn;
 
-        if (sendEth) _unwrapWETHTo(payable(msg.sender), l1TokensToReturn);
-        else IERC20(l1Token).safeTransfer(msg.sender, l1TokensToReturn);
-
+        if (sendEth) {
+            weth.withdraw(l1TokensToReturn);
+            payable(msg.sender).transfer(l1TokensToReturn);
```

Comment:
```suggestion
            payable(msg.sender).transfer(l1TokensToReturn); // This will revert if the caller is a contract that does not implement a fallback function.
```